pkg/pool/pool.go

```diff
@@ -59,6 +60,7 @@ func (pb *Builder) Build(ctx context.Context, options *BuilderOptions) (Pool, er
 
 // Pool is an interface providing connection artifacts on request.
 type Pool interface {
+	client.Object
```

Comment:
I think we should also add some
```
var _ Pool = (*pool)(nil)
```
to ensure that `pool` always implements `Pool` completely.